src/mbgl/text/placement.cpp

```diff
@@ -538,8 +538,12 @@ void Placement::placeSymbolBucket(const BucketPlacementData& params, std::set<ui
             // Erase it so that the placement result from the non-fading tile supersedes it
             placements.erase(symbolInstance.crossTileID);
         }
-        
-        placements.emplace(symbolInstance.crossTileID, JointPlacement(placeText || alwaysShowText, placeIcon || alwaysShowIcon, offscreen || bucket.justReloaded));
+
+        auto pair = placements.emplace(
+            symbolInstance.crossTileID,
+            JointPlacement(placeText || alwaysShowText, placeIcon || alwaysShowIcon, offscreen || bucket.justReloaded));
+        assert(pair.second);
+        newSymbolPlaced(symbolInstance, pair.first->second, placementType, textBoxes, iconBoxes);
```

Comment:
We will have a vtable lookup on this critical code path that is only used for debugging. :-(

Reply:
we could move the check `if (collectData) ...` here and have an `assert(collectData)` in the `::newSymbolPlaced()` implementation.

Reply:
I think compilers do better job optimizing vtable lookup than `if (...)`